enginetest/queries.go

```diff
@@ -4699,6 +4715,14 @@ var QueryTests = []QueryTest{
 			{1, 1, 4},
 		},
 	},
+	{
+		Query: `SELECT /*+ JOIN_ORDER(mytable, othertable) */ s2, i2, i FROM mytable INNER JOIN (SELECT * FROM othertable) othertable ON i2 = i`,
```

Comment:
Should have a test with a join condition with more than one column that you expect to get used by the hash lookup cache

This might be covered by another test in here, but better to just add one you're sure exercises the logic